config/servicemappings/bigqueryconnection.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: core.cnrm.cloud.google.com/v1alpha1
+kind: ServiceMapping
+metadata:
+  name: bigqueryconnection.cnrm.cloud.google.com
+  namespace: cnrm-system
+spec:
+  name: BigQueryConnection
+  version: v1beta1
+  serviceHostName: bigqueryconnection.googleapis.com
+  resources:
+    - name: google_bigquery_connection
+      kind: BigQueryConnectionConnection
+      v1alpha1ToV1beta1: true
+      storageVersion: v1alpha1
+      iamConfig:
+        policyName: google_bigquery_connection_iam_policy
+        policyMemberName: google_bigquery_connection_iam_member
+        referenceField:
+          name: connection_id
+          type: name
+        supportsConditions: false
+      iamMemberReferenceConfig:
```

Comment:
Yes! This would need extra work. I've created a separate bug on KCC: **b/339349950**- KCC to support retrieval of service account from bigquery connections to be referenced in iamPolicyMember

Reply:
Gotcha, thank you for the context! I'd suggest you split this feature after we support this resource (so that we can control the size and scope of the PR).